export docs (#4)
phero-hq/website: docs/deploying-phero/nodejs.md

---
title: "Deploying a NodeJS server"
---

> In order to deploy your Phero server, there's one last step: making an export.
> When you export you server, Phero compiles an optimized server for you to deploy.

Reviewer: ```suggestion
In order to deploy your Phero server, there's one last step: making an export. When you export your server, Phero compiles an optimised bundle for you to deploy.
```

> It does **not** use any framework like `express` or `fastify`. It compiles down to plain JavaScript which uses the built-in `http` module from NodeJS itself. This way we can guarantee there's little to no overhead between an issued request and actually calling your function.

Reviewer: ```suggestion
It does **not** use any framework like `express` or `fastify`. It compiles down to plain JavaScript which uses the built-in `http` module from NodeJS itself. This way we can guarantee that there's as little overhead as possible between a request and actually calling your function.
```

## Steps

> First navigate to your server dir:

Reviewer: ```suggestion
First, navigate to your server directory:
```

```
cd api
```

To export as a NodeJS server, run:

```
npx phero@latest server export --flavor nodejs
```

> This will generate a node script for each of your services. To run one of your services:

Reviewer: ```suggestion
This will generate a node script for each of your services in an `export` directory. To run one of these services:
```

```
  cd ./export/helloWorldService
  npm i
  node index.js
```

> That's it! From here it should be fairly straight forward to deploy it to serveral cloud providers or anywhere you like.

Reviewer: ```suggestion
That's it! From here you can deploy this bundle to any provider that supports NodeJS.
```
